### PR TITLE
DOC: make legend docstring interpolated

### PR DESCRIPTION
**Pull request by jklymak**
## PR Summary

The three legend docstrings (`legend.py`, `ax.legend`, `fig.legend`) have the same "Other Parameters" section.  These had drifted apart over the years (as had their acceptable kwargs).  We unified them #9324, *but* there was some protest about using the docstring interp method, so I cut and pasted the docstrings, and added a test to `test_legend` that made sure the strings were identical.

Not surprisingly, this has caused problems.  The test routinely fails when someone has edited one of the three docstrings.  Good, its catching inconsistencies, but its a bit hard to tell folks to cut and paste the same thing three time.

I (somewhat strongly) think docstring interpolation is the better option here.  Faced with the mysteries of `%(_legend_kw_doc)s` I think its save to assume the naive user will search the code base for this string and see thats where to put their new description of (yet another) kwarg for legend.  

Could be convinced that just referencing `legend.__init__` would be OK, but I do like the kwargs to appear in the local docstring.  

## PR Checklist

- [x] Code is PEP 8 compliant
- [x] Documentation is sphinx and numpydoc compliant

### CODE REVIEW

**Comment by jklymak** [comment]
https://7779-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/legend_api.html

Does "Other Parameters" not behave the same as Parameters?  Or is this an indenting issue?

**Comment by jklymak** [comment]
https://7779-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/_as_gen/matplotlib.axes.Axes.legend.html#matplotlib.axes.Axes.legend

https://7779-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/_as_gen/matplotlib.figure.Figure.html#matplotlib.figure.Figure

**Comment by jklymak** [comment]
OTOH, this looks the same as the current docs, so I'll argue is an improvement.  

https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.legend.html#matplotlib.axes.Axes.legend

**Comment by timhoffm** [comment]
> Does "Other Parameters" not behave the same as Parameters? Or is this an indenting issue?

It should behave the same. Probably an indenting issue.

**Comment by jklymak** [comment]
OK, well, I'll try to hack until its fixed...

**Comment by jklymak** [comment]
@timhoffm  when I had this stuff hardcoded in (before this change) it also failed to look the same as Parameters.  So, I'm not sure this PR isn't OK as is, though I'd be happy to fix the issue if its easy to fix.

**Comment by timhoffm** [comment]
Are we talking about the same thing?

https://7779-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/legend_api.html has issues with the handles and labels parameters, whereas

https://matplotlib.org/devdocs/api/legend_api.html?highlight=legend#module-matplotlib.legend_handler

Does not.

**Comment by jklymak** [comment]
I agree with that.  But it had issues before this PR as well.  Ie look at the current version of the docs (linked above).   

So the problem isn’t the docstring interpolation. 

OTOH I’m willing to fix the problem, I just have no idea what it is

**Comment by timhoffm** [comment]
@jklymak Do you mean that the grey bar extends to the right instead of down? This is probably a bug in numpydoc. See #6924.

**Comment by jklymak** [comment]
@timhoffm Thats indeed what I meant.  

This PR doesn't make the formatting worse, and makes the docstring currating substantially easier.  The "Other Parameters" bug is a different battle.

File: lib/matplotlib/legend.py [hunk 1]
```diff
@@ -364,173 +364,7 @@ def __init__(self, parent, handles, labels,
...
-            The spacing between columns.
-            Measured in font-size units.
-            Default is ``None``, which will take the value from
-            :rc:`legend.columnspacing`.
-
-        handler_map : dict or None
-            The custom dictionary mapping instances or types to a legend
-            handler. This `handler_map` updates the default handler map
-            found at :func:`matplotlib.legend.Legend.get_legend_handler_map`.
+        %(_legend_kw_doc)s
```
Comment (tacaswell) [inline lib/matplotlib/legend.py]: https://7895-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/legend_api.html#matplotlib.legend.Legend

vs

https://matplotlib.org/api/legend_api.html#matplotlib.legend.Legend

Something is not right here, the handles and labels parameters are being extra indented, but I can not see why....

Comment (tacaswell) [reply lib/matplotlib/legend.py]: https://matplotlib.org/devdocs/api/legend_api.html#matplotlib.legend.Legend devdocs also look ok....

Comment (jklymak) [reply lib/matplotlib/legend.py]: I see what you and @timhoffm mean now - you mean in the `Parameter` section *above* this change.  Silly me, only checking the part of the doctoring I edited...

I cannot see why either.  Very strange as the `Axes.legend` and `Figure.legend` docstrings are fine, and I don't see how they are any different than this doctoring...

File: lib/matplotlib/legend.py [hunk 2]
```diff
@@ -348,7 +348,6 @@ def __init__(self, parent, handles, labels,
                  handler_map=None,
                  ):
         """
-
```
Comment (jklymak) [inline lib/matplotlib/legend.py]: This extra space was causing the hiccup in the formatting.  Sphinx is pretty fragile....

**Comment by jklymak** [comment]
OK, fixed:  https://7929-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/api/legend_api.html
There was an extra carriage return...

**Pull request merged**